megamek/src/megamek/common/MULParser.java

```diff
@@ -1832,6 +1846,89 @@ private void parseAeroCrit(Element aeroCritTag, Entity entity){
     }
     
     /**
+     *  Parse a dropCrit tag for the given <code>Entity</code>.
+     *  @param dropCritTag
+     *  @param entity
+     */
+    private void parseDropCrit(Element dropCritTag, Entity entity){
+    	String dockingcollar = dropCritTag.getAttribute(DOCKING_COLLAR);
+    	String kfboom = dropCritTag.getAttribute(KFBOOM);
+    	
+    	Dropship d = (Dropship) entity;
+    	
+    	if (dockingcollar.length() > 0) {
+    		d.setDamageDockCollar(true);
+    	}
+    	if (kfboom.length() > 0) {
+    		d.setDamageKFBoom(true);
+    	}
+    }
+    
+    /**
+     *  Parse cargo bay and door the given <code>Entity</code>.
+     *  Borrowed all this from the code that handles vehicle stabilizer crits by location.
+     *  
+     *  @param entity
+     */
+    private void parseTransportBay (Element bayTag, Entity entity) {
+    	// Look for the element's attributes.
+    	String index = bayTag.getAttribute(INDEX);
+    	
+    	int bay;
+    	// Did we find the required index?
+    	if ((index == null) || (index.length() == 0)) {
+    		warning.append("Could not find index for bay.\n");
+    		return;
+    	} else {
+    	// Try to get a good index value.
+    		bay = -1;
+    		try {
+    			bay = Integer.parseInt(index);
+    		} catch (NumberFormatException excep) {
+    			// Handled by the next if test
+    		}
+    		if (bay < 0) {
+    			warning.append("Found invalid index value for bay: ").append(index).append(".\n");
+    			return;
+    		} else if (bay > entity.getTransportBays().size()) {
+    			warning.append("The entity, ")
+    			.append(entity.getShortName())
+    			.append(" does not have a bay at index: ")
+    			.append(bay).append(".\n");
+    			return;
+    		}
+    	} // End check for required fields
+    	
+    	Bay currentbay = entity.getBayById(bay);
+    	
+    	// Handle children for each bay.
+    	NodeList nl = bayTag.getChildNodes();
+    	for (int i = 0; i < nl.getLength(); i++) {
+    		Node currNode = nl.item(i);
+    		
+    		if (currNode.getParentNode() != bayTag) {
+    			continue;
+    		}
+    		int nodeType = currNode.getNodeType();
+    		if (nodeType == Node.ELEMENT_NODE) {
+    			Element currEle = (Element) currNode;
+    			String nodeName = currNode.getNodeName();
+    			if (nodeName.equalsIgnoreCase(BAYDAMAGED)) {
+    				String bayhit = currEle.getAttribute(DAMAGED);
+    				if (bayhit.equals("true")) {
+    					currentbay.setbayDamaged();
+    				}
+    			}
+    			if (nodeName.equalsIgnoreCase(BAYDOORS)) {
+    				String currentdoors = currEle.getAttribute(DOORS);
+    				int doors = Integer.parseInt(currentdoors);
+    				currentbay.setCurrentDoors(doors);
+    			}
+    		}
+    	}
+    } // End parseTransportBay
```

Comment:
The indentation looks wrong in this block of code.  It looks too deep.  Either you've got double tabs, or the tabs are set to 8 spaces.  Are you using spaces or tabs?  We should have spaces only, inserting 4 spaces for a tab.

Reply:
I have my tab set to 4 spaces. Is there a vertical ruler I can use to make sure things line up?

Reply:
If you have the code formatter configured to use spaces and not tabs, and it's configure for 4 spaces, you can use the code formatter.  If you highlight a block of code, you can right click and I think it's Source > Format (shortcut if ctrl-shift-F).  If you don't have anything highlighted it will refactor the whole file (possibly not what you want).

This assumes you're using Eclipse.